src/libos/src/fs/timer_file.rs

```diff
@@ -0,0 +1,174 @@
+use super::*;
+
+use crate::time::{itimerspec_t, timespec_t};
+use atomic::{Atomic, Ordering};
+use std::time::Duration;
+
+/// Native Linux timerfd
+#[derive(Debug)]
+pub struct TimerFile {
+    host_fd: HostFd,
+    host_events: Atomic<IoEvents>,
+    notifier: IoNotifier,
+}
+
+impl TimerFile {
+    pub fn new(clockid: u32, flags: TimerCreationFlags) -> Result<Self> {
+        let raw_host_fd = try_libc!({
```

Comment:
> EINVAL
The clockid argument is neither CLOCK_MONOTONIC nor CLOCK_REALTIME;

Please check the clock_id before ocall, we have defined ClockID in src/time/mod.rs

Reply:
Done

Reply:
I have not seen  the check